src/main/java/uk/co/ramp/EvaluateCase.java

```diff
@@ -0,0 +1,230 @@
+
+                break;
+            case REQUESTED_TEST:
+                LOGGER.trace("user {} is awaiting test result", p.id());
+                updateAlertStatus(AWAITING_RESULT, time);
+                p.setWasInfectious(p.isInfectious());
+                break;
+            case ALERTED:
+                updateAlertStatus(REQUESTED_TEST, time);
+                LOGGER.trace("user {} has requested a test. Current Status: {}-{}", p.id(), p.status(), p.alertStatus());
+                break;
+            case NONE:
+                LOGGER.trace("user {} has been alerted. Current Status: {}-{}", p.id(), p.status(), p.alertStatus());
+                updateAlertStatus(ALERTED, time);
+                break;
+        }
+
+        return new HashSet<>();
+    }
+
+    private Set<Integer> alertAllContacts() {
+
+        Set<Integer> contactIds = new HashSet<>();
+        // add all ids
+        for (ContactRecord r : p.contactRecords()) {
+            contactIds.add(r.from());
+            contactIds.add(r.to());
+        }
+
+        // remove self
+        contactIds.remove(p.id());
+
+        LOGGER.trace("Alerting " + contactIds);
+        return contactIds;
+
+    }
+
+
+    private void updateAlertStatus(final AlertStatus newStatus, final int currentTime) {
+
+        int time = getDistributionValue(properties.meanTestTime(), properties.maxTestTime());
+        AlertStatus oldstatus = p.alertStatus();
+        p.setAlertStatus(newStatus);
+        p.setNextAlertStatusChange(currentTime + time);
+        LOGGER.trace("id: {} alertStatus: {} -> {} at t={}. Current Virus status: {}", p.id(), oldstatus, newStatus, currentTime, p.status());
+
+    }
+
+    private VirusStatus determineOutcome(Case p) {
+        //TODO add real logic
+        return p.health() > rng.nextUniform(0, 1) ? RECOVERED : DEAD;
+    }
+
+    private VirusStatus determineInfection(Case p) {
+        //TODO add real logic
+        return p.health() > rng.nextUniform(0, 1) ? INFECTED : INFECTED_SYMP;
+    }
+
+    public void randomExposure(int t) {
+        p.setExposedBy(-2);
+        p.setExposedTime(t);
+        if (p.status() == SUSCEPTIBLE) {
+            LOGGER.trace("Person with id: {} has been randomly exposed at time {}", p.id(), t);
+            updateVirusStatus(EXPOSED, t);
+        } else {
+            String message = String.format("The person with id: %d should not be able to transition from %s to %s", p.id(), p.status(), EXPOSED);
+            LOGGER.error(message);
+            throw new InvalidStatusTransitionException(message);
+        }
+    }
+
+    int getDistributionValue(double mean, double max) {
+
+        int value = (int) Math.round(mean);
+        double sample;
+        switch (properties.progressionDistribution()) {
+            case GAUSSIAN:
+                sample = rng.nextGaussian(mean, mean / 2d);
+                break;
+            case LINEAR:
+                sample = rng.nextUniform(mean - mean / 2d, mean + mean / 2d);
```

Comment:
Hi @sam-brett , can you explain why we are generating a sample between `0.5*mean` and `1.5*mean` for LINEAR instead of from `mean - (max - mean)` to `max`? We are not able to create a uniform distribution from 0 to 1 using this method using the current approach

Reply:
That’ll just be a hangover from when the mean was the only variable.